### ISSUE TITLE
Product Collection: allow inserting collections contextually

### ISSUE THREAD

**Issue by kmanijak** [woothemes/woocommerce#47966]
**Issue Description:**

In #47964, we plan to implement API which will allow collections to define required context. By required context, we mean the context required by collection to work properly. Each collection will be able to define the required context. These are the context/location types we defined (that list may be extended in the future):

- **Site** – generic, default.
- **Product** – single product template or block.
- **Order** – order confirmation template.
- **Cart** – cart/checkout templates or cart/checkout/mini-cart blocks.
- **Archive** – an archive page related to WC, such as the `product_cat` or `product_tag` taxonomies.

Now, There are two cases to consider:
- Case 1 (Preview Label): When the required context isn't available on Editor side but will be available in Frontend. In this case, we can show a preview label (https://github.com/woocommerce/woocommerce/issues/47964) 
- Case 2: When required context won't be available on Editor as well as Frontend. In this case, we shouldn't show the collection in the collection chooser & block inserter. If someone copy pastes the collection in Editor or insert it using some other way, then it should show nothing on Frontend for now.

> [!IMPORTANT]
> **This issue is to address case 2 as a followup to https://github.com/woocommerce/woocommerce/issues/47964.**

In future, we will implemented better ways to handle missing context in 2nd case. See #44878, #44877, #44879 for more info.

**Goal:**
- Handle 2nd case as described above.
	- Don't show the collection in the collection chooser & block inserter.
	- If someone copy pastes the collection in Editor or insert it using some other way, then it should show nothing on Frontend for now. And should show "Preview Label" in Editor.

Context visibility matrix
<img width="793" alt="image" src="https://github.com/woocommerce/woocommerce/assets/20098064/f5afd8b2-f40e-4d61-9bb0-1fe3e76bda93">

Ref: zx83mG5ZnDiacPdH6GlO2j-fi

**Comment by gigitux** [comment]
@kmanijak should we keep this issue open?

**Comment by kmanijak** [comment]
No, there's no plan on addressing it anytime soon. I'm closing it as "not planned".
